### ISSUE TITLE
feature request: Add locations field to Menu

### ISSUE THREAD

**Issue by TylerBarnes** [wp-graphql/wp-graphql#1135]
Currently there's no way to take a list of Menu nodes and programatically determine which one is assigned to which location. A `locations` field will be helpful here. Without it the new Gatsby source plugin won't have this info

**Comment by TylerBarnes** [comment]
I think to go along with this, omitting the location where arg on the `menuItems` field should give you a list of all existing MenuItem nodes

**Comment by rodrigo-arias** [comment]
I found the same thing, I can filter by location from WordPress but not from Gatsby. It would be great to have it

**Comment by jasonbahl** [comment]
closed by #1348 

Re: locations

We can query for locations like so: 

```
{
  menus {
    nodes {
      id
      name
      locations
      menuItems {
        nodes {
          id
          label
          locations
        }
      }
    }
  }
}
```

And get a result like so: 

```json
{
  "data": {
    "menus": {
      "nodes": [
        {
          "id": "dGVybTo1Mw==",
          "name": "Another Menu",
          "locations": [
            "EXPANDED",
            "FOOTER",
            "SOCIAL"
          ],
          "menuItems": {
            "nodes": [
              {
                "id": "bmF2X21lbnVfaXRlbTo2MzQ=",
                "label": "Test Page",
                "locations": [
                  "EXPANDED",
                  "FOOTER",
                  "SOCIAL"
                ]
              },
              {
                "id": "bmF2X21lbnVfaXRlbTo2MzU=",
                "label": "Spanish Page",
                "locations": [
                  "EXPANDED",
                  "FOOTER",
                  "SOCIAL"
                ]
              },
              {
                "id": "bmF2X21lbnVfaXRlbTo2MzY=",
                "label": "English Page",
                "locations": [
                  "EXPANDED",
                  "FOOTER",
                  "SOCIAL"
                ]
              }
            ]
          }
        },
        {
          "id": "dGVybTo1NA==",
          "name": "DELETE ME",
          "locations": null,
          "menuItems": {
            "nodes": [
              {
                "id": "bmF2X21lbnVfaXRlbTo2Mzg=",
                "label": "DELETE ME",
                "locations": null
              }
            ]
          }
        },
        {
          "id": "dGVybTo0NQ==",
          "name": "Test",
          "locations": [
            "PRIMARY"
          ],
          "menuItems": {
            "nodes": [
              {
                "id": "bmF2X21lbnVfaXRlbTo2Mjk=",
                "label": "Sample Page",
                "locations": [
                  "PRIMARY"
                ]
              },
              {
                "id": "bmF2X21lbnVfaXRlbTo2MzA=",
                "label": "Child Page",
                "locations": [
                  "PRIMARY"
                ]
              },
              {
                "id": "bmF2X21lbnVfaXRlbTo2MzE=",
                "label": "Grandchild",
                "locations": [
                  "PRIMARY"
                ]
              },
              {
                "id": "bmF2X21lbnVfaXRlbTo1NTE=",
                "label": "test",
                "locations": [
                  "PRIMARY"
                ]
              },
              {
                "id": "bmF2X21lbnVfaXRlbTo1NTI=",
                "label": "Test Page",
                "locations": [
                  "PRIMARY"
                ]
              }
            ]
          }
        },
        {
          "id": "dGVybTo1NQ==",
          "name": "Unpublished Menu",
          "locations": null,
          "menuItems": {
            "nodes": [
              {
                "id": "bmF2X21lbnVfaXRlbTo2Mzk=",
                "label": "Test Page",
                "locations": null
              }
            ]
          }
        }
      ]
    }
  }
}
```

In action in GraphiQL:

![Screen Shot 2020-06-24 at 9 20 01 PM](https://user-images.githubusercontent.com/1260765/85649468-7a1c1d80-b660-11ea-92f1-9d184a928bfe.png)

----

Re: 
> I think to go along with this, omitting the location where arg on the menuItems field should give you a list of all existing MenuItem nodes

The default now for querying menuItems from the root is to return _all_ menu items. 

Take this query for example: 

```graphql
{
  menuItems {
    nodes {
      id
      label
      locations
      menu {
        node {
          name
        }
      }
    }
  }
}
```

We will get results like below, which include menu items across different menus and assigned to different locations: 

```json
{
  "data": {
    "menuItems": {
      "nodes": [
        {
          "id": "bmF2X21lbnVfaXRlbTo2MzQ=",
          "label": "Test Page",
          "locations": [
            "EXPANDED",
            "FOOTER",
            "SOCIAL"
          ],
          "menu": {
            "node": {
              "name": "Another Menu"
            }
          }
        },
        {
          "id": "bmF2X21lbnVfaXRlbTo2Mjk=",
          "label": "Sample Page",
          "locations": [
            "PRIMARY"
          ],
          "menu": {
            "node": {
              "name": "Test"
            }
          }
        },
        {
          "id": "bmF2X21lbnVfaXRlbTo2MzU=",
          "label": "Spanish Page",
          "locations": [
            "EXPANDED",
            "FOOTER",
            "SOCIAL"
          ],
          "menu": {
            "node": {
              "name": "Another Menu"
            }
          }
        },
        {
          "id": "bmF2X21lbnVfaXRlbTo2MzA=",
          "label": "Child Page",
          "locations": [
            "PRIMARY"
          ],
          "menu": {
            "node": {
              "name": "Test"
            }
          }
        },
        {
          "id": "bmF2X21lbnVfaXRlbTo2MzY=",
          "label": "English Page",
          "locations": [
            "EXPANDED",
            "FOOTER",
            "SOCIAL"
          ],
          "menu": {
            "node": {
              "name": "Another Menu"
            }
          }
        },
        {
          "id": "bmF2X21lbnVfaXRlbTo2MzE=",
          "label": "Grandchild",
          "locations": [
            "PRIMARY"
          ],
          "menu": {
            "node": {
              "name": "Test"
            }
          }
        },
        {
          "id": "bmF2X21lbnVfaXRlbTo1NTE=",
          "label": "test",
          "locations": [
            "PRIMARY"
          ],
          "menu": {
            "node": {
              "name": "Test"
            }
          }
        },
        {
          "id": "bmF2X21lbnVfaXRlbTo1NTI=",
          "label": "Test Page",
          "locations": [
            "PRIMARY"
          ],
          "menu": {
            "node": {
              "name": "Test"
            }
          }
        }
      ]
    }
  }
}
```

In Action in GraphiQL:

![Screen Shot 2020-06-24 at 9 22 59 PM](https://user-images.githubusercontent.com/1260765/85649650-e39c2c00-b660-11ea-8946-34ea2764845f.png)
